src/classes/STG_PanelRDHealthCheck_TEST.cls

```diff
@@ -0,0 +1,285 @@
+        STG_PanelRDHealthCheck healthCheck = new STG_PanelRDHealthCheck(ctrl);
+
+        // Verify known packaged installment periods on this field
+        // =============================================================
+        Set<String> supportedInstallmentPeriods = new Set<String>();
+        supportedInstallmentPeriods.add(RD_Constants.INSTALLMENT_PERIOD_WEEKLY);
+        supportedInstallmentPeriods.add(RD_Constants.INSTALLMENT_PERIOD_MONTHLY);
+        supportedInstallmentPeriods.add(RD_Constants.INSTALLMENT_PERIOD_QUARTERLY);
+        supportedInstallmentPeriods.add(RD_Constants.INSTALLMENT_PERIOD_YEARLY);
+        supportedInstallmentPeriods.add(RD_Constants.INSTALLMENT_PERIOD_FIRST_AND_FIFTEENTH);
+
+        healthCheck.validateSupportedPicklistValues(supportedInstallmentPeriods, 'npe03__Installment_Period__c',
+            System.Label.healthDetailsInvalidPicklistValue, System.Label.healthSolutionUnsupportedPicklistValue);
+
+        healthCheck.validateRequiredPicklistValues(supportedInstallmentPeriods, 'npe03__Installment_Period__c',
+            System.Label.healthDetailsMissingRequiredPicklistValue, System.Label.healthSolutionMissingPicklistValue);
+
+        String results = getValidationResults(ctrl);
+        System.assertEquals(0, ctrl.listDR.size(),
+            'There should zero errors in this picklist validation\n' + results
+        );
+
+        // Add an invalid "required" picklist to this list and validate that the required
+        // validation logs the error.
+        // =============================================================
+        supportedInstallmentPeriods.add(BAD_PICKLIST_VALUE);
+        healthCheck.validateRequiredPicklistValues(supportedInstallmentPeriods, 'npe03__Installment_Period__c',
+            System.Label.healthDetailsInvalidPicklistValue, System.Label.healthSolutionUnsupportedPicklistValue);
+
+        results = getValidationResults(ctrl);
+        System.assertEquals(1, ctrl.listDR.size(),
+            'There should exactly one error in this picklist validation.\n' + results
+        );
+        System.assert(results.contains(BAD_PICKLIST_VALUE),
+            'The error should reference the "' + BAD_PICKLIST_VALUE + '" picklist value.\n' + results
+        );
+    }
+
+    /**
+     * @description Validate that the checking of the Opportunity Stage Settings works as expected
+     */
+    @IsTest
+    private static void shouldValidateOppStageSettings() {
+        STG_PanelHealthCheck_CTRL ctrl = new STG_PanelHealthCheck_CTRL();
+        STG_PanelRDHealthCheck healthCheck = new STG_PanelRDHealthCheck(ctrl);
+
+        healthCheck.validateOpportunityStage(PLEDGED_STAGE, false, false);
+
+        String results = getValidationResults(ctrl);
+        System.assertEquals(0, ctrl.listDR.size(),
+            'There should zero errors in the record type validation\n' + results
+        );
+
+        healthCheck.validateOpportunityStage(PLEDGED_STAGE, true, true);
```

Comment:
This seems beginning of another concept, should it be extracted into another unit test?
Also, I think we should name methods to convey what is being validated without providing input Booleans. For example, we can add into the `STG_PanelRDHealthCheck` and verify those calls from tests? If not let's provide vars to store Boolean input args?
```

    /**
     * @description Validates specified stage name exists as an open stage and it is active
     * @param oppStageName Opportunity Stage Name
     * @return void
     */
    private void validateOpenOpportunityStage(String oppStageName) {
        validateOpportunityStage(oppStageName, false, false);
    }

    /**
     * @description Validates specified stage name exists as a Closed Lost stage and it is active
     * @param oppStageName Opportunity Stage Name
     * @return void
     */
    private void validateClosedLostOpportunityStage(String oppStageName) {
        validateOpportunityStage(oppStageName, false, true);
    }
```